src/main/java/com/icthh/xm/uaa/service/UserService.java

```diff
@@ -650,8 +676,42 @@ public User updateLastLoginDate(User user) {
         return userRepository.save(user);
     }
 
+    public void onSuccessfulLogin(String userKey) {
+        Integer maxPasswordAttempts = tenantPropertiesService.getTenantProps().getSecurity().getMaxPasswordAttempts();
+
+        userRepository.findOneByUserKey(userKey).ifPresent(user -> {
+            if (applicationProperties.isLastLoginDateEnabled()) {
+                user.updateLastLoginDate();
+            }
+
+            if (maxPasswordAttempts != null && maxPasswordAttempts > 0) {
+                user.resetPasswordAttempts();
+            }
+        });
+    }
+
+    @LogicExtensionPoint(value = "IncreaseFailedPasswordAttempts")
+    public void increaseFailedPasswordAttempts(String username) {
+        Integer maxPasswordAttempts = tenantPropertiesService.getTenantProps().getSecurity().getMaxPasswordAttempts();
+
+        if (maxPasswordAttempts != null && maxPasswordAttempts > 0) {
+            userLoginRepository.findOneByLoginIgnoreCase(username)
+                .map(UserLogin::getUser)
+                .ifPresent(user -> {
+                    user.incrementPasswordAttempts();
+                    self.disableAccountActivation(user, maxPasswordAttempts);
+                });
+        }
+    }
+
     public void handlePasswordReset(PasswordResetRequest resetRequest) {
         passwordResetHandlerFactory.getPasswordResetHandler(resetRequest.getResetType()).handle(resetRequest);
     }
 
+    @LogicExtensionPoint(value = "DisableAccountActivation")
```

Comment:
this lep should  be invoked only when user.getPasswordAttempts().equals(maxPasswordAttempts) is true

Reply:
DisableAccountActivation it's separate lep that should be invoken only when you really disable user, not every failed tries

Reply:
and pls rename it to PasswordAttemptsExceeded

Reply:
Yes, you are absolutely right. Renamed